CMakeLists.txt

```
@@ -1,71 +1,55 @@
 cmake_minimum_required(VERSION 3.2)
 project(binapi)
 
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 20)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -fsanitize=address")
+find_package(Threads REQUIRED)
+find_package(OpenSSL REQUIRED SSL Crypto)
+find_package(ZLIB REQUIRED)
 
-add_definitions(
-    -UNDEBUG
-    -DDTF_HEADER_ONLY
-)
+add_library(binapi SHARED
+        src/api.cpp
+        src/dtf.cpp
+        src/enums.cpp
+        src/errors.cpp
+        src/pairslist.cpp
+        src/reports.cpp
+        src/tools.cpp
+        src/types.cpp
+        src/websocket.cpp)
 
-include_directories(
-    ./include
-)
+target_include_directories(binapi PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
-if (DEFINED ${BOOST_INCLUDE_DIR})
-    include_directories(
-        ${BOOST_INCLUDE_DIR}
-    )
-endif()
+target_compile_options(binapi PUBLIC
```

```suggestion
target_compile_options(binapi PRIVATE
```
You probably don’t want to propagate these options to users.

Also, `-s` is a linker option, I'm not sure it has any effect here. Better to move to `target_link_options`.